pkg/websocket/websocket.go

```diff
@@ -1,7 +1,10 @@
 package websocket
 
+import "github.com/surrealdb/surrealdb.go/internal/rpc"
+
 type WebSocket interface {
 	Connect(url string) (WebSocket, error)
 	Send(method string, params []interface{}) (interface{}, error)
 	Close() error
+	LiveNotifications(id string) (chan rpc.RPCResponse, error)
```

Comment:
The return type shouldnt be an rpc response (an implementation detail). It should be a `Notification` type, that you would need to add to the golang driver.
This is what it looks like:
https://github.com/surrealdb/surrealdb/blob/main/lib/src/dbs/notification.rs#L24-L28

Reply:
right this makes sense however i am not so sure how it would work in the codebase. What I mean is that in `gorilla.go` in the `initialize` function, we are sending all the responses to response channel which is `rpc.RPCResponse` type.
If we make a new `Notification` type in rpc we wont be able to write to the channel. 
This would mean we would need to write to another channel which is of the type Notification, eventually adding 2 more functions like the `createReponseChannel` and `getResponseChannel`  for the websocket. Also addition of notificationChannel in the `WebSocket` struct.

Do i understand it correctly?

Reply:
Ah right yes, you are right there - this is relaying protocol to driver. In that case the protocol should be agnostic - this should instead be "Receive async message". That's because in the future, it would be more than one type of message. Addtionally, a web socket doesn't "know" what a live query is at that abstraction level. So you want to avoid mixing layers.